lib/src/over_react_test/react_util.dart

```diff
@@ -446,12 +451,17 @@ Element getComponentRootDomByTestId(dynamic root, String value, {String key = de
 ///     queryByTestId(renderedInstance, 'value'); // returns the `inner` `<div>`
 ///
 /// Related: [queryAllByTestId], [getComponentRootDomByTestId].
-Element queryByTestId(dynamic root, String value, {String key = defaultTestIdKey}) {
-  return findDomNode(root).querySelector('[$key~="$value"]');
+Element queryByTestId(dynamic root, String value, {String key = defaultTestIdKey, bool searchInShadowDom = false, int shadowDepth}) {
+  var results = _findDeep(findDomNode(root), _makeTestIdSelector(value, key: key), searchInShadowDom: searchInShadowDom, findMany: false, depth: shadowDepth);
+  return results.isNotEmpty ? results.first : null;
 }
 
 /// Returns all descendant [Element]s of [root] that has their [key] html attribute value set to [value].
 ///
+/// Setting [searchInShadowDom] to true will allow the query to search within ShadowRoots that use `mode:"open"`.
+/// [shadowDepth] will limit how many layers of ShadowDOM will searched, by default it will seach infinately deep, this
+/// should only be needed if there is alot of ShadowRoots within ShadowRoots.
```

Comment:
Same as  above
```suggestion
/// [shadowDepth] will limit how many layers of ShadowDOM will searched. By default it will search infinitely deep, and this
/// should only be needed if there are a lot of ShadowRoots within ShadowRoots.
```